assets/js/blocks/reviews-by-product/block.js

```diff
@@ -12,63 +12,121 @@ import PropTypes from 'prop-types';
 import { renderReview } from './utils';
 import withComponentId from '../../utils/with-component-id';
 
+export const getReviews = ( args ) => {
+	return apiFetch( {
+		path: '/wc/blocks/products/reviews?' + Object.entries( args ).map( ( arg ) => arg.join( '=' ) ).join( '&' ),
+		parse: false,
+	} ).then( ( response ) => {
+		return response.json().then( ( reviews ) => {
+			const totalReviews = parseInt( response.headers.get( 'x-wp-total' ), 10 );
+			return { reviews, totalReviews };
+		} );
+	} );
+};
+
 /**
  * Component to handle edit mode of "Reviews by Product".
  */
 class ReviewsByProduct extends Component {
 	constructor() {
 		super( ...arguments );
 		const { attributes } = this.props;
+		const { order, orderby } = this.getOrderArgs( attributes.orderby );
+
 		this.state = {
-			orderby: attributes.orderby,
+			order,
+			orderby,
 			reviews: [],
 			totalReviews: 0,
 		};
 
 		this.onChangeOrderby = this.onChangeOrderby.bind( this );
-		this.getReviews = this.getReviews.bind( this );
 		this.appendReviews = this.appendReviews.bind( this );
 	}
 
 	componentDidMount() {
-		this.getReviews();
+		this.loadFirstReviews();
 	}
 
 	componentDidUpdate( prevProps ) {
```

Comment:
In a follow-up I would like to split this component so we don't load all the `componentDidUpdate` logic in the frontend, since attributes can't change there.